Examples/Modules/laser_injection/analysis_2d.py

```diff
@@ -0,0 +1,199 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.signal import hilbert
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+import checksumAPI
+
+# Maximum acceptable error for this test
+relative_error_threshold = 0.05
+
+# A small number
+small_num = 1.0e-8
+
+# Physical parameters
+um = 1.e-6
+fs = 1.e-15
+c = 299792458
+
+# Parameters of the gaussian beam
+wavelength = 1.*um
+w0 = 5.*um
+tt = 10.*fs
+x_c = 10.*um
+t_c = 20.*fs
+# foc_dist = 13.109*um (not actually used)
+E_max = 4e12
+
+# laser direction
+dir_vector = np.array([2.,0,1.0])
+dir_vector /= np.linalg.norm(dir_vector)
+
+rot_angle = np.arctan(dir_vector[2]/dir_vector[0])
+
+# polarization vector
+pol_vector = np.array([1.0,1.0,-2.0])
+pol_vector /= np.linalg.norm(pol_vector)
+
+# Simulation parameter
+steps=220
+
+# Calculates the envelope of a Gaussian beam
+def gauss_env(T,XX,ZZ):
+    '''Function to compute the theory for the envelope
+    '''
+
+    X =  np.cos(rot_angle)*(XX-x_c) + np.sin(rot_angle)*ZZ
+    Z = -np.sin(rot_angle)*(XX-x_c) + np.cos(rot_angle)*ZZ
+
+    inv_tau2 = 1./tt/tt
+    inv_w_2 = 1.0/(w0*w0)
+    exp_arg = - (Z*Z)*inv_w_2 - inv_tau2 / c/c * (X-T*c)*(X-T*c)
+    return E_max * np.real(np.exp(exp_arg))
+
+# Checks envelope and central frequency for a given laser component
+def check_component(data, component, t_env_theory, coeff, X,Z,dx,dz):
+    print("*** Checking " + component + " ***")
+    field = data['boxlib', component].v.squeeze()
+    env = abs(hilbert(field))
+
+    env_theory = t_env_theory*np.abs(coeff)
+
+    # Plot results
+    fig = plt.figure(figsize=(12,6))
+
+    ax1 = fig.add_subplot(221, aspect='equal')
+    ax1.set_title('PIC field')
+    p1 = ax1.pcolormesh(X,Z,field)
+    cax1 = make_axes_locatable(ax1).append_axes('right', size='5%', pad=0.05)
+    fig.colorbar(p1, cax=cax1, orientation='vertical')
+
+    ax2 = fig.add_subplot(222, aspect='equal')
+    ax2.set_title('PIC envelope')
+    p2 = ax2.pcolormesh(X,Z,env)
+    cax2 = make_axes_locatable(ax2).append_axes('right', size='5%', pad=0.05)
+    fig.colorbar(p2, cax=cax2, orientation='vertical')
+
+    ax3 = fig.add_subplot(223, aspect='equal')
+    ax3.set_title('Theory envelope')
+    p3 = ax3.pcolormesh(X,Z,env_theory)
+    cax3 = make_axes_locatable(ax3).append_axes('right', size='5%', pad=0.05)
+    fig.colorbar(p3, cax=cax3, orientation='vertical')
+
+    ax4 = fig.add_subplot(224, aspect='equal')
+    ax4.set_title('Difference')
+    p4 = ax4.pcolormesh(X,Z,env-env_theory)
+    cax4 = make_axes_locatable(ax4).append_axes('right', size='5%', pad=0.05)
+    fig.colorbar(p4, cax=cax4, orientation='vertical')
+
+    plt.tight_layout()
+    plt.savefig("plt_" + component + ".png", bbox_inches='tight')
+
+    if( np.abs(coeff) < small_num):
+        is_field_zero = np.sum(np.abs(env)) < small_num
+        print("[OK] This field component is ~ 0.0")
+        assert(is_field_zero)
```

Comment:
Is there a reason for printing `OK` before actually doing the assert?

Reply:
ehm...no actually. It's a mistake...

Reply:
I should have fixed that